paparazzi/src/main/java/app/cash/paparazzi/PaparazziSdk.kt

```diff
@@ -309,16 +326,54 @@ public class PaparazziSdk @JvmOverloads constructor(
         (view.parent as ViewGroup).removeView(view)
       }
       AnimationHandler.sAnimatorHandler.set(null)
+
       if (hasComposeRuntime) {
         forceReleaseComposeReferenceLeaks()
       }
     }
   }
 
-  private fun withTime(
+  private fun renderForIdleRecomposer(
+    recomposer: Any?,
+    frame: Int,
+    additionalRenderCount: Int,
+    onAdditionalRender: () -> Result
+  ): Result {
+    val result = onAdditionalRender()
+
+    if (result.status == ERROR_UNKNOWN) {
+      throw result.exception
+    }
+
+    if (hasComposeRuntime) {
+      // If there is a recomposition that needs to happen, we need to trigger it within the context of the first frame.
+      val recomposerInstance = recomposer as? Recomposer
+      val hasPendingWork = recomposerInstance?.hasPendingWork
+
+      if (frame < frameLimit && hasPendingWork == true) {
```

Comment:
Yep, this does the same logic but allows for customization of what frames you want to retry.
Happy to remove this for simplification

Reply:
> allows for customization of what frames you want to retry.

yea, this is indeed what I'd like to avoid.  To be clear, this pending recomposition trick is already a hack that Android Studio has decided to allow given how the Compose framework (and more specifically, in this example, Popup) requires another layout pass.  We could take a stance of not supporting it, but I imagine that would hurt adoption.

That being said, customization would allow developers to work around improperly-designed UI elements that are skipping multiple frames before they finally draw to the Canvas.  I am fairly Android Studio would not support as that would be a developer error, and neither should we.

Reply:
> Happy to remove this for simplification

Yes, let's remove/inline renderForIdleRecomposer since its more handling of an edge case vs a general-purpose solution.